codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectAggregator.java

```diff
@@ -151,7 +151,11 @@ protected boolean isAggregated(HttpObject msg) throws Exception {
 
     @Override
     protected boolean isContentLengthInvalid(HttpMessage start, int maxContentLength) {
-        return getContentLength(start, -1L) > maxContentLength;
+        try {
+            return getContentLength(start, -1L) > maxContentLength;
+        } catch (final NumberFormatException e) {
+            return false;
```

Comment:
Because `true` here means that the content length exceeds the max content length, that's not the case if the content length is invalid, so false.